File: tugas12/apiclient/Anggota.py
import requests
import json
class Anggota:
    def __init__(self):
        self.__id=None
        self.__kode_anggota = None
        self.__nim = None
        self.__nama_anggota = None
        self.__jk = None
        self.__prodi = None
        self.__alamat = None
        self.__url = "http://localhost/appperpus/anggota_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def kode_anggota(self):
        return self.__kode_anggota
        
    @kode_anggota.setter
    def kode_anggota(self, value):
        self.__kode_anggota = value
    @property
    def nim(self):
        return self.__nim
        
    @nim.setter
    def nim(self, value):
        self.__nim = value
    @property
    def nama_anggota(self):
        return self.__nama_anggota
        
    @nama_anggota.setter
    def nama_anggota(self, value):
        self.__nama_anggota = value
    @property
    def jk(self):
        return self.__jk
        
    @jk.setter
    def jk(self, value):
        self.__jk = value
    @property
    def prodi(self):
        return self.__prodi
        
    @prodi.setter
    def prodi(self, value):
        self.__prodi = value
    @property
    def alamat(self):
        return self.__alamat
        
    @alamat.setter
    def alamat(self, value):
        self.__alamat = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_nim(self, nim):
        url = self.__url+"?nim="+nim
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id_anggota']
            self.__kode_anggota = item['kode_anggota']
            self.__nim = item['nim']
            self.__nama_anggota = item['nama_anggota']
            self.__jk = item['jk']
            self.__prodi = item['prodi']
            self.__alamat = item['alamat']
        return data
    def simpan(self):
        payload = {
            "kode_anggota":self.__kode_anggota,
            "nim":self.__nim,
            "nama_anggota":self.__nama_anggota,
            "jk":self.__jk,
            "prodi":self.__prodi,
            "alamat":self.__alamat
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_nim(self, nim):
        url = self.__url+"?nim="+nim
        payload = {
            "kode_anggota":self.__kode_anggota,
            "nim":self.__nim,
            "nama_anggota":self.__nama_anggota,
            "jk":self.__jk,
            "prodi":self.__prodi,
            "alamat":self.__alamat
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_nim(self,nim):
        url = self.__url+"?nim="+nim
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text

File: tugas12/appperpus/Anggota.php
<?php
//Simpanlah dengan nama file : Anggota.php
require_once 'database.php';
class Anggota 
{
    private $db;
    private $table = 'anggota';
    public $kode_anggota = "";
    public $nim = "";
    public $nama_anggota = "";
    public $jk = "";
    public $prodi = "";
    public $alamat = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_nim(int $nim)
    {
        $query = "SELECT * FROM $this->table WHERE nim = $nim";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`kode_anggota`,`nim`,`nama_anggota`,`jk`,`prodi`,`alamat`) VALUES ('$this->kode_anggota','$this->nim','$this->nama_anggota','$this->jk','$this->prodi','$this->alamat')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET kode_anggota = '$this->kode_anggota', nim = '$this->nim', nama_anggota = '$this->nama_anggota', jk = '$this->jk', prodi = '$this->prodi', alamat = '$this->alamat' 
        WHERE id_anggota = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_nim($nim): int
    {
        $query = "UPDATE $this->table SET kode_anggota = '$this->kode_anggota', nim = '$this->nim', nama_anggota = '$this->nama_anggota', jk = '$this->jk', prodi = '$this->prodi', alamat = '$this->alamat' 
        WHERE nim = $nim";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id_anggota = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_nim($nim): int
    {
        $query = "DELETE FROM $this->table WHERE nim = $nim";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>